ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayFeatureLayers/DisplayFeatureLayers.cpp

```diff
@@ -0,0 +1,232 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "DisplayFeatureLayers.h"
+
+#include "Envelope.h"
+#include "Error.h"
+#include "FeatureLayer.h"
+#include "Geodatabase.h"
+#include "GeodatabaseFeatureTable.h"
+#include "GeoPackage.h"
+#include "GeoPackageFeatureTable.h"
+#include "LayerListModel.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MapTypes.h"
+#include "Point.h"
+#include "Portal.h"
+#include "PortalItem.h"
+#include "ServiceFeatureTable.h"
+#include "ShapefileFeatureTable.h"
+#include "SpatialReference.h"
+#include "Viewpoint.h"
+
+#include <QStandardPaths>
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace
+{
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::HomeLocation);
+#endif
+
+  return dataPath + "/ArcGIS/Runtime/Data/";
+}
+} // namespace
+
+DisplayFeatureLayers::DisplayFeatureLayers(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(BasemapStyle::ArcGISTopographic, this))
+{
+}
+
+DisplayFeatureLayers::~DisplayFeatureLayers() = default;
+
+void DisplayFeatureLayers::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DisplayFeatureLayers>("Esri.Samples", 1, 0, "DisplayFeatureLayersSample");
+}
+
+MapQuickView* DisplayFeatureLayers::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void DisplayFeatureLayers::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  emit mapViewChanged();
+}
+
+void DisplayFeatureLayers::setLayerMode(FeatureLayerMode featureLayerMode)
+{
+  if (!m_map)
+    return;
+
+  m_map->operationalLayers()->clear();
+
+  m_currentLayerMode = featureLayerMode;
+
+  switch (featureLayerMode) {
+    case FeatureLayerMode::Geodatabase:
+      setGeodatabaseLayer();
+      break;
+    case FeatureLayerMode::Geopackage:
+      setGeopackageLayer();
+      break;
+    case FeatureLayerMode::PortalItem:
+      setPortalItemLayer();
+      break;
+    case FeatureLayerMode::ServiceFeatureTable:
+      setServiceFeatureTableLayer();
+      break;
+    case FeatureLayerMode::Shapefile:
+      setShapefileLayer();
+      break;
+    default:
+      break;
+  }
+}
+
+void DisplayFeatureLayers::setGeodatabaseLayer()
+{
+  Geodatabase* gdb = new Geodatabase(defaultDataPath() + "geodatabase/LA_Trails.geodatabase", this);
+  connect(gdb, &Geodatabase::doneLoading, this, [gdb, this](Error e)
```

Comment:
I checked our code base and we overwhelmingly use `Error` without const or &. I can update that if we want?

Reply:
let's update this code since it is a new sample, then log a follow up issue to update everything. All value objects in signals are passed as const & in v200